backend/fault.py

```diff
@@ -12,53 +11,39 @@ def __init__(self, message):
 
 
 class Fault:
-    def __init__(self, tlc, severity, rank, level, suffix, okValue, faultValue,
-                 rack, description, name):
+    def __init__(self, tlc, severity, suffix, okValue, faultValue, description,
+                 name, prefix):
         self.tlc = tlc
         self.severity = int(severity)
-        self.rank = rank
-        self.level = level
-        self.rack = rack
-        self.suffix = suffix
         self.description = description
         self.name = name
         self.okValue = float(okValue) if okValue else None
         self.faultValue = float(faultValue) if faultValue else None
 
+        self.pv: PV = PV(prefix + ":" + suffix, connection_timeout=PV_TIMEOUT)
```

Comment:
I'm not very familiar with this type of in line commenting (not sure what to call this actually?). What's happening here? The colon in "self.pv: " is the source of my confusion

Reply:
Oh it's a type annotation! I'm telling pycharm that self.pv is a PV (like, I could've done self.faultValue: float = ...

Reply:
Ah, okay! That makes sense:D Is this type of type annotation specific to pycharm or will any editor/compiler be able to read this as well? 

(I suppose we'll find out quickly as soon as we run it in production if the OPIs can or cannot)

Reply:
They should be able to run without compilation issues, it's just _only_ useful in IDEs haha So the OPIs won't get any benefit, but they won't have any problems. Also it just helps for readability even if it doesn't do anything, so I think it's a win/win